packages/iceworks-server/src/lib/plugin/project-manager/app.ts

```diff
@@ -418,7 +433,7 @@ class ProjectManager extends EventEmitter {
    */
   public async deleteProject(params: { projectPath: string; deleteFiles?: boolean }): Promise<void> {
     const { projectPath, deleteFiles } = params;
-    this.projects = this.projects.filter(({ path }) => path !== projectPath);
+    this.projects = this.projects && this.projects.filter(({ path }) => path !== projectPath);
```

Comment:
Under what conditions would it be undefined?

Reply:
When the currently selected project in `iceworks` is deleted from the local environment before `iceworks-server`is started, `this.projects` would be `undefined`. So we need to check it here.

Reply:
`this.projects` define by `await this.refresh()`, I think should alway be a array.

Reply:
If  the currently selected project in iceworks is deleted from the local environment, you've already dealt with it at `getCurrent`.